### PR TITLE
LPS-162979 Avoid circular redirect 

### PR DESCRIPTION
**Pull request by ces-hongvo**
Hi,
This is the PR for the ticket https://issues.liferay.com/browse/LPS-162979. Please help to review it and leave your comments.
Thank you.

### CODE REVIEW

**Comment by ces-binhtran** [comment]
Try to use _`layoutFriendlyURLLocalService.getLayoutFriendlyURLs(layout.getPlid())`_ to check whether the redirect and original request have the same layout.

**Comment by ces-hongvo** [comment]
Hi @ces-binhtran 
I have updated this PR based on your suggestions. Could you please double-check?
Thank you.

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java [hunk 1]
```diff
@@ -751,6 +760,55 @@ private SiteFriendlyURL _getAlternativeSiteFriendlyURL(
...
+
+				List<LayoutFriendlyURL> layoutFriendlyURLs =
+					layoutFriendlyURLLocalService.getLayoutFriendlyURLs(
+						layout.getPlid(), destinationLayoutFriendlyURL, 0, 1);
+
+				if (ListUtil.isNotEmpty(layoutFriendlyURLs)) {
+					LayoutFriendlyURL redirectedLayoutFriendlyURL =
+						layoutFriendlyURLs.get(0);
+
+					redirectedLayoutFriendlyURL.getFriendlyURL();
```
Comment (ces-binhtran) [inline modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java]: Redundant

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java [hunk 2]
```diff
@@ -751,6 +760,55 @@ private SiteFriendlyURL _getAlternativeSiteFriendlyURL(
...
+
+					Locale redirectedLocale = LocaleUtil.fromLanguageId(
+						redirectedLayoutFriendlyURL.getLanguageId());
+
+					String redirect = portal.getLocalizedFriendlyURL(
+						httpServletRequest, layout, redirectedLocale, locale);
+
+					boolean forcePermanentRedirect = _isPermanentRedirect(
+						group.getCompanyId());
```
Comment (ces-binhtran) [inline modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java]: Only use 1 time...inline

**Comment by ces-hongvo** [comment]
### Root cause summary
When we set a URL redirect to a translated URL of the same layout. It will try to redirect back to the current language URL
https://github.com/ces-binhtran/liferay-portal/blob/d85fe3a9cc6e7bf419e5219c1ba14d9b7f0780d7/modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java#L328
### 
Steps to reproduce
Please see the https://issues.liferay.com/browse/LPS-162979

### Steps to reproduce explained
Create a Layout then set the redirect to its translated URL (any language)

### Solution notes
When founding a redirect, I will try to check if the redirect URL is related to the current Layout. If so, I will create a redirectURL which also updates the locale of the browser to be the new URL's locale. It will prevent the circular redirect back to the old URL's locale.

**Comment by ces-binhtran** [comment]
https://github.com/liferay-lima/liferay-portal/pull/3852

**Pull request closed without merging**